blueocean-config/src/main/java/io/jenkins/blueocean/config/BlueOceanConfigFactory.java

```diff
@@ -0,0 +1,45 @@
+package io.jenkins.blueocean.config;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+
+/**
+ * Blue Ocean configuration factory which allows different extensions to return configuration values.
+ * 
+ */
+public abstract class BlueOceanConfigFactory implements ExtensionPoint {
```

Comment:
Please move it to `io.jenkins.blueocean.rest.model.factory` package in blueocean-rest module.